From 8aaa8703a3ef45c289ff6db9eba1e7acffa12073 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Mon, 20 Jan 2020 18:17:58 +0700
Subject: [PATCH 630/630] media: vsp1: Add mutex lock for stopping DRM
 pipelines

This patch is to avoid possible race condition when stopping DRM
pipelines.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/media/platform/vsp1/vsp1_drm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/vsp1/vsp1_drm.c b/drivers/media/platform/vsp1/vsp1_drm.c
index 302d02a..f34dd6c 100644
--- a/drivers/media/platform/vsp1/vsp1_drm.c
+++ b/drivers/media/platform/vsp1/vsp1_drm.c
@@ -90,9 +90,11 @@ int vsp1_du_setup_lif(struct device *dev, unsigned int width,
 		/* Zero width or height means the CRTC is being disabled, stop
 		 * the pipeline and turn the light off.
 		 */
+		mutex_lock(&pipe->lock);
 		ret = vsp1_pipeline_stop(pipe);
 		if (ret == -ETIMEDOUT)
 			dev_err(vsp1->dev, "DRM pipeline stop timeout\n");
+		mutex_unlock(&pipe->lock);
 
 		media_entity_pipeline_stop(&pipe->output->entity.subdev.entity);
 
-- 
2.7.4

